### ISSUE TITLE
LAB: SQL Injection GET instead of POST?

### ISSUE THREAD

**Issue by stebiger** [WebGoat/WebGoat#253]
Hey,

I am currently looking at the sql injection lab and was noticing that the stage 1 - 3 all use a HTTP GET request instead of a POST.
Therefour the learned injection attack is not working - is there something broken with webgoat 7 or is it on purpose?
Unfortunately the solution can also not be displayed.

**Comment by misfir3** [comment]
With some of the modernization we've done with WebGoat, it's technically by design, but has some implications such as this.  There is a way to make it work as a POST per Lesson.  Also something we plan to carry forward and improve.  Which SQLi lesson exactly is this on?  I know the standard String SQL Injection lesson works (as I used it recently).

Thanks for reporting it!

**Comment by stebiger** [comment]
I don't see a "real" problem by having it intended to be a GET request - unfortunately I have no idea how to bypass it then with a get parameter - I think encoding of the value plays a role. 
Yet I am unsure if it is intended to be that way or not

**Comment by misfir3** [comment]
True ... still, we want to provide the capability to allow lessons to set the method for submission while maintaining ease of authoring (main focus of our efforts in new development right now)

**Comment by nbaars** [comment]
@stebiger: tested the bypass of the login and it still works, indeed it is no longer a POST but a GET. You can change the password when you tamper the data (don't use a space)

**Comment by nbaars** [comment]
Fixed the hints and remove the spaces.
@stebiger I'm still able to bypass the login page when I intercept the 'GET' request and copy & paste the hint.

**Comment by stebiger** [comment]
@nbaars thanks - after thinking about the way the get works and will encode parameters I was able to solve the lab.
Anyhow thanks for working on the issue :-)

**Comment by lvernaillen** [comment]
Actually, this problem is not restricted to the SQL Injection lab.
**ALL** forms I checked are using GET although the form tag specifies method="POST"!
I attached a screenshot proving it for the very first lesson: "General - Http Basics".
But it has this issue for the other forms as well.

If I add a simple html file with a form with post method, it works as expected. Only the Webgoat pages containing forms don't work. Something in the framework you use? Are other events capturing the POST and changing it into GET?

![form_method_get_instead_of_post](https://cloud.githubusercontent.com/assets/4180374/18289115/61d94a36-747e-11e6-93b1-b75b0fd80526.png)
